### PR TITLE
Add validation specs to `CustomFilter` model

### PR DESCRIPTION
**Pull request by mjankowski**
This contains JUST the specs + tiny refactor from https://github.com/mastodon/mastodon/pull/27602 - but not the marshall format change or normalizes changes.

### CODE REVIEW

File: app/models/custom_filter.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def filter_action=(value)
   accepts_nested_attributes_for :keywords, reject_if: :all_blank, allow_destroy: true
 
   validates :title, :context, presence: true
-  validate :context_must_be_valid
+  validate :context_validation_error, if: :invalid_context_value?
```
Comment (ClearlyClaire) [inline app/models/custom_filter.rb:57]: I'm not sure about this refactor, I expect the validation function to actually validate the value, not unconditionally add an error…

**Review by ClearlyClaire — COMMENTED** [review]
The specs are good but I'm not sure about the refactor

File: app/models/custom_filter.rb (hunk 1 above)
Comment (mjankowski) [reply app/models/custom_filter.rb:57]: Is your objection the moving to two methods at all, or the moving of the conditional up into the `validate` options?

Comment (mjankowski) [reply app/models/custom_filter.rb]: Updated to keep the conditional in the validation method; but kept the extraction of the "valid check" method.

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**